### ISSUE TITLE
text2image segfault when using --list_available_fonts

### ISSUE THREAD

**Issue by mgeerdsen** [tesseract-ocr/tesseract#2009]
### Environment

* **Tesseract Version**: 4.0.0-rc3-41-g0a42c0
* **Platform**: Ubuntu 18.04 Docker container (Linux 8196cc005e68 4.9.0-8-amd64 #1 SMP Debian 4.9.110-3+deb9u6 (2018-10-08) x86_64 x86_64 x86_64 GNU/Linux)

### Current Behavior:
using 4.0.0-rc3-41-g0a42c0
```
root@8196cc005e68:~# text2image --list_available_fonts --fonts_dir /opt/fonts

(process:514): Pango-CRITICAL **: 14:08:44.619: pango_font_family_get_name: assertion 'PANGO_IS_FONT_FAMILY (family)' failed
Segmentation fault (core dumped)
```

### Expected Behavior:
using 4.0.0-beta.4
```
root@c05d6b14efe4:/# text2image --list_available_fonts --fonts_dir /opt/fonts
  0: Alte Schwabacher
```

Probably related to this... as mentioned in #1999 I am also experiencing blank rendered tiffs in this version, whereas 4.0.0-beta.4 shows regular tiffs containing the text. This is already true for the sample_text.txt.tif as created using tesseract_training.sh.

**Comment by mgeerdsen** [comment]
bisecting confirms that this has been introduced in d1d73b98888285744caf6ddaaccb3963594e5acf

**Comment by Shreeshrii** [comment]
@zdenop I can confirm the blank rendered tifs with current version. Training process is broken with this, so it should be fixed before 4.0.0.

To reproduce, try the following:

```
 ~/tesseract/src/training/tesstrain.sh \
>   --fonts_dir /usr/share/fonts \
>   --lang eng --linedata_only \
>   --noextract_font_properties \
>   --langdata_dir ../langdata \
>   --tessdata_dir ../tessdata \
>   --fontlist "Impact Condensed" \
>   --training_text ../langdata_lstm/eng/eng.training_text \
>   --workspace_dir ~/tmp/ \
>   --maxpages 3 \
>   --save_box_tiff \
>   --output_dir ../tesstutorial/engeval

=== Starting training for language 'eng'
[Sat Oct 20 03:50:00 UTC 2018] /home/ubuntu/tesseract/src/training/text2image --fonts_dir=/usr/share/fonts --font=Impact Condensed --outputbase=/home/ubuntu/tmp//fc-cache/sample_text.txt --text=/home/ubuntu/tmp//fc-cache/sample_text.txt --fontconfig_tmpdir=/home/ubuntu/tmp//fc-cache
Stripped 1 unrenderable words
Rendered page 0 to file /home/ubuntu/tmp//fc-cache/sample_text.txt.tif

=== Phase I: Generating training images ===
Rendering using Impact Condensed
[Sat Oct 20 03:50:01 UTC 2018] /home/ubuntu/tesseract/src/training/text2image --fontconfig_tmpdir=/home/ubuntu/tmp//fc-cache --fonts_dir=/usr/share/fonts --strip_unrenderable_words --leading=32 --char_spacing=0.0 --exposure=0 --outputbase=/tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0 --max_pages=3 --font=Impact Condensed --text=../langdata_lstm/eng/eng.training_text
Stripped 530 unrenderable words
Rendered page 0 to file /tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0.tif
Stripped 524 unrenderable words
Rendered page 1 to file /tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0.tif
Stripped 532 unrenderable words
Rendered page 2 to file /tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0.tif

=== Phase UP: Generating unicharset and unichar properties files ===
[Sat Oct 20 03:50:04 UTC 2018] /home/ubuntu/tesseract/src/training/unicharset_extractor --output_unicharset /tmp/tmp.1QCfba4yrr/eng.unicharset --norm_mode 1 /tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0.box
Failed to read data from: /tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0.box
Wrote unicharset file /tmp/tmp.1QCfba4yrr/eng.unicharset
[Sat Oct 20 03:50:04 UTC 2018] /home/ubuntu/tesseract/src/training/set_unicharset_properties -U /tmp/tmp.1QCfba4yrr/eng.unicharset -O /tmp/tmp.1QCfba4yrr/eng.unicharset -X /tmp/tmp.1QCfba4yrr/eng.xheights --script_dir=../langdata
Loaded unicharset of size 3 from file /tmp/tmp.1QCfba4yrr/eng.unicharset
Setting unichar properties
Setting script properties
Writing unicharset to file /tmp/tmp.1QCfba4yrr/eng.unicharset

=== Phase E: Generating lstmf files ===
Using TESSDATA_PREFIX=../tessdata
[Sat Oct 20 03:50:04 UTC 2018] /home/ubuntu/tesseract/src/api/tesseract /tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0.tif /tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0 --psm 6 lstm.train
Tesseract Open Source OCR Engine v4.0.0-rc3-46-g94b76 with Leptonica
Page 1
Page 2
Page 3
ERROR: /tmp/tmp.1QCfba4yrr/eng.Impact_Condensed.exp0.lstmf does not exist or is not readable


```

Do you want a separate issue to be opened for it?

**Comment by Shreeshrii** [comment]
I can also confirm the crash:

```
ubuntu@tesseract-ocr:~/tesstutorial$ text2image --list_available_fonts --fonts_dir /usr/share/fonts

(process:5033): Pango-CRITICAL **: pango_font_family_get_name: assertion 'PANGO_IS_FONT_FAMILY (family)' failed
Segmentation fault (core dumped)
ubuntu@tesseract-ocr:~/tesstutorial$ text2image -v
4.0.0-rc3-46-g94b76

```

**Comment by amitdo** [comment]
Another option:
#736 (comment)
Should be tested on all supported platforms.

You can replace putenv with _putenv
https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/putenv-wputenv?view=vs-2017

**Comment by zdenop** [comment]
should be  fixed at 1b2bda65e039952af9c1336ff36761fa241c2df4. Please test.

**Comment by mgeerdsen** [comment]
Thanks, I can confirm that it is working again, listing fonts and rendering text to images both.
